### PR TITLE
[Headers]: Add headers docs

### PR DESCRIPTION
**Pull request by maya**
## Description

Resubmitting #1409 against `staging`. 

Adds documentation to headers component.

https://standards-staging-add-headers-docs.apps.cloud.gov/headers/

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

File: docs/_components/headers/headers.html [hunk 1]
```diff
@@ -3,82 +3,158 @@
...
+  <li>Include skip navigation links to allow users with screen readers to bypass long navigation lists. Make sure you include an id at the beginning of your main content and that it matches the skipnav link. Find more information here: <a href="http://webaim.org/techniques/skipnav/">http://webaim.org/techniques/skipnav/</a></li>
+  <li>Include Tab focus for all top-level navigation navigation items — this will allow keyboard-reliant users to easily navigate interactive items.</li>
+  <li>Ensure your horizontal navigation is keyboard compatible; test to make sure users can use Tab to navigate and Space (or Enter) to open pages.</li>
+  <li>Avoid using hover to expand dropdown lists. Hover is difficult for some users and won't work on touch screens. Drop downs should expand on click or with keyboard navigation.</li>
+  <li>We use semantic header and nav elements with <code>role="banner"</code> and <code>role="navigation"</code> respectively. <code>role="banner"</code> is your masthead.</li>
+  <li>You can use multiple nav elements for groups of navigation links, but you should only use one <code>role=”navigation”</code> for the main nav of a page.</li>
```
Comment (juliaelman) [inline docs/_components/headers/headers.html]: The quotes around `"role=navigation"` are a different typeset than the others. Might be a copy/paste error.

File: docs/_components/headers/headers.html [hunk 2]
```diff
@@ -3,82 +3,158 @@
...
+</ul>
 
 <h2 class="usa-heading" id="basic">Basic header</h2>
 <div class="preview">
   <a class="media_link" href="{{ site.baseurl }}/headers/basic/">
-    <img src="{{ site.baseurl }}/assets/img/headers/header-basic.png" alt="">
+    <img src="{{ site.baseurl }}/assets/img/headers/header-basic.png" alt="Basic header">
   </a>
-  <a class="usa-button" href="{{ site.baseurl }}/headers/basic/">Demo basic header</a>
+  <a class="usa-button" href="{{ site.baseurl }}/headers/basic/" alt="">Demo basic header</a>
```
Comment (juliaelman) [inline docs/_components/headers/headers.html]: Missing alt text.

Comment (maya) [inline docs/_components/headers/headers.html]: Removed in f224300.

File: docs/_components/headers/headers.html (hunk 1 above)
Comment (maya) [inline docs/_components/headers/headers.html]: Fixed in c39c6dd.

**Pull request merged**